artio-codecs/src/main/java/uk/co/real_logic/artio/fields/ReadOnlyDecimalFloat.java

```diff
@@ -0,0 +1,295 @@
+     * Get the number of digits to the right of the decimal point.
+     *
+     * @return the number of digits to the right of the decimal point.
+     */
+    public int scale()
+    {
+        return this.scale;
+    }
+
+    public void appendTo(final StringBuilder builder)
+    {
+        CodecUtil.appendFloat(builder, this);
+    }
+
+    @Override
+    public String toString()
+    {
+        final StringBuilder builder = new StringBuilder();
+        appendTo(builder);
+        return builder.toString();
+    }
+
+    @Override
+    public final int compareTo(final ReadOnlyDecimalFloat other)
+    {
+        final long value = this.value;
+        final int scale = this.scale;
+
+        final long otherValue = other.value;
+        final int otherScale = other.scale;
+
+        final long decimalPointDivisor = PowerOf10.pow10(scale);
+        final long otherDecimalPointDivisor = PowerOf10.pow10(otherScale);
+
+        final long valueBeforeDecimalPoint = value / decimalPointDivisor;
+        final long otherValueBeforeDecimalPoint = otherValue / otherDecimalPointDivisor;
+
+        final int beforeDecimalPointComparison = Long.compare(valueBeforeDecimalPoint, otherValueBeforeDecimalPoint);
+
+        if (beforeDecimalPointComparison != 0)
+        {
+            // Can be determined using just the long value before decimal point
+            return beforeDecimalPointComparison;
+        }
+
+        // values after decimal point, but has removed scale entirely
+        long valueAfterDecimalPoint = (value % decimalPointDivisor);
+        long otherValueAfterDecimalPoint = (otherValue % otherDecimalPointDivisor);
+
+        // re-normalise with scales by multiplying the lower scale number up
+        if (scale > otherScale)
+        {
+            final int differenceInScale = scale - otherScale;
+            otherValueAfterDecimalPoint *= PowerOf10.pow10(differenceInScale);
+        }
+        else
+        {
+            final int differenceInScale = otherScale - scale;
+            valueAfterDecimalPoint *= PowerOf10.pow10(differenceInScale);
+        }
+
+        return Long.compare(valueAfterDecimalPoint, otherValueAfterDecimalPoint);
+    }
+
+    /**
+     * This method should satisfy the Effective Java - 3rd Edition - Methods Common to All Objects Item 10
+     * "Obey the general contract when overriding equals",
+     * and it should do it for every x and y where x and y can be an instance of
+     * ReadOnlyDecimalFloat or DecimalFloat, in any order, this method is:
+     * Reflexive : x.equal(x) return true
+     * Symmetric : if x.equals(y) return true, y.equals(x) return true
+     * Transitive : if x.equals(y) return true and y.equals(z) return true, x.equals(z) return true
+     * Consistent : invocations of x.equals(y) return the same value
+     * For any x != null, x.equals(null) returns false
+     * <p>
+     * Any class other than DecimalFloat inheriting ReadOnlyDecimalFloat should overwrite this equal method.
```

Comment:
This method is final - is that possible?

Reply:
Good spot, I contradicted myself. I think at some point I realised that we actually don't want anyone to extend this class and forgot to delete the comment.